### ISSUE TITLE
TalkerFlutter.init() not printing logs onto console with iOS 16

### ISSUE THREAD

**Issue by arhamsc** [Frezyx/talker#139]
**Describe the bug**
When developing on physical iOS device, TalkerFlutter.init() is not printing logs onto the console.

**Smartphone (please complete the following information):**
 - Device: iPhone 13 Pro Max
 - OS: 16.6
 - Flutter Version: 3.13.0 (stable)

**Additional context**
When i shifted to just Talker(), it worked but prints special characters before and after the message - `<....>`

**Comment by nikhil7299** [comment]
@Frezyx I was using talker_logger to log Riverpod 2 state changes using the Provider Observer State Logger from official Riverpod 2 docs. Where I realized that iOS builds does not show colors in debug console. So I internally replaced the "print" with "devtools.log". Then It started working.

**Comment by arhamsc** [comment]
@nikhil7299 in Talker_Flutter.init(), the log method for iOS is debug log, as per my inspection of the package. If you could share which file you changed it would help me produce a similar outcome as yours.

**Comment by nikhil7299** [comment]
I just replaced print to devtools.log inside logger.dart in talker_logger like this -
Previous Code :-
`import 'package:talker_logger/talker_logger.dart';

class TalkerLogger {
  TalkerLogger({
    this.settings = const TalkerLoggerSettings(),
    LoggerFilter? filter,
    this.formatter = const ExtendedLoggerFormatter(),
    void Function(String message)? output,
  }) {
    // ignore: avoid_print
    _output =
        output ?? (String message) => message.split('\n').forEach(print);
    _filter = filter ?? LogLevelFilter(settings.level);
    ansiColorDisabled = false;
  }
...
`
New Code :-
`import 'package:talker_logger/talker_logger.dart';
import 'dart:developer' as devtools;

class TalkerLogger {
  TalkerLogger({
    this.settings = const TalkerLoggerSettings(),
    LoggerFilter? filter,
    this.formatter = const ExtendedLoggerFormatter(),
    void Function(String message)? output,
  }) {
    // ignore: avoid_print
    _output =
        output ?? (String message) => message.split('\n').forEach(devtools.log);
    _filter = filter ?? LogLevelFilter(settings.level);
    ansiColorDisabled = false;
  }
...
`
=> Also run in debug mode as devtool log works in debug mode.

**Comment by nikhil7299** [comment]
I just replaced print to devtools.log inside logger.dart in talker_logger like this -

Previous Code :-
```
import 'package:talker_logger/talker_logger.dart';

class TalkerLogger {
  TalkerLogger({
    this.settings = const TalkerLoggerSettings(),
    LoggerFilter? filter,
    this.formatter = const ExtendedLoggerFormatter(),
    void Function(String message)? output,
  }) {
    // ignore: avoid_print
    _output =
        output ?? (String message) => message.split('\n').forEach(print);
    _filter = filter ?? LogLevelFilter(settings.level);
    ansiColorDisabled = false;
  }
...
```

New Code :-
```
import 'package:talker_logger/talker_logger.dart';
import 'dart:developer' as devtools;

class TalkerLogger {
  TalkerLogger({
    this.settings = const TalkerLoggerSettings(),
    LoggerFilter? filter,
    this.formatter = const ExtendedLoggerFormatter(),
    void Function(String message)? output,
  }) {
    // ignore: avoid_print
    _output =
        output ?? (String message) => message.split('\n').forEach(devtools.log);
    _filter = filter ?? LogLevelFilter(settings.level);
    ansiColorDisabled = false;
  }
...
```
=> Also run in debug mode as devtool log works in debug mode.

**Comment by arhamsc** [comment]
Did you check TalkerFlutter.init(), source code? It is actually similar to this. But when I used that started for a few times it  was printing fine but suddenly it stopped printing.

**Comment by arhamsc** [comment]
@Frezyx 
Sorry for the late reply, but the code is as follows:
```
 _talker = TalkerFlutter.init();
```

I tried doing what @nikhil7299 did, the devtools.log is not working actually, when i reverted it back to 'print' it worked, but with the special characters as mentioned earlier without colors.

**Comment by nikhil7299** [comment]
@arhamsc Did you run in debug mode or without debugging? Developer tools .log() works only in debug mode.

**Comment by arhamsc** [comment]
> @arhamsc Did you run in debug mode or without debugging? Developer tools .log() works only in debug mode.

I ran the app in debug mode. And then tried changing back and forth from developer log and print, only with print it worked not with developer logs.

**Comment by arhamsc** [comment]
Is there any update on this thread? @Frezyx

**Comment by RNOVOSELOV** [comment]
@Frezyx  @arhamsc 
I also reproduced  same problem.
 
In VSCode logs are displayed correctly, but Android Studio (Giraffe version) does not  show the logs or shows them without color and not completely (tested on iOS 15 & iOS16) 

Android and Web are works correctly on both IDE.

**Comment by RNOVOSELOV** [comment]
```
final sl = GetIt.instance;

void _setupTakerLogger() {
  sl.registerLazySingleton<Talker>(() => TalkerFlutter.init());
  Bloc.observer = TalkerBlocObserver(talker: sl.get<Talker>());
  FlutterError.onError =
      (details) => sl.get<Talker>().handle(details.exception, details.stack);
}
```
Android Studio's console log example, when run on iOS 15

> [Talker] [38;5;4m┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────[0m
> [38;5;4m│ [INFO] | 14:43:25 763ms | _onSplashPageLoaded - Read token from shared preferences: null[0m
> [38;5;4m└──────────────────────────────────────────────────────────────────────────────────────────────────────────────[0m
>

**Comment by nikhil7299** [comment]
@arhamsc  
Yes I am talking about 'flutter run --debug' but do not use command line, just press F5, the logs will be shown in the Debug Console and not in Terminal.
Also when you modify the internal code of TalkerLogger, make sure to reload "VS Code" window and re-run your flutter app using F5.

**Comment by arhamsc** [comment]
@nikhil7299 Yes I used the internal vs code run command. And I have tried the steps you have mentioned. It seems to be the general problem of the apple making internal changes which aren't allowing the logs in general within flutter framework.

**Comment by SharbelOkzan** [comment]
Any updates/workaround?

**Comment by arhamsc** [comment]
It seems to be printing onto IoS console in a neater way now.

Device: Iphone 14 Pro Max
IOS: 17

But there is not color coding and has some weird alignment and symbols.
